test/remote/gateways/remote_authorize_net_test.rb

```diff
@@ -74,6 +74,48 @@ def test_successful_authorize_and_capture
     assert_success capture
   end
 
+  def test_base_test_mode_does_not_send_a_test_request
+    original_base_mode = ActiveMerchant::Billing::Base.mode
+    ActiveMerchant::Billing::Base.mode = :test
+
+    gateway = AuthorizeNetGateway.new(fixtures(:authorize_net))
+    response = gateway.purchase(@amount, @credit_card, @options)
+
+    assert_success response
+    assert_equal('0', response.params['test_request'])
+  ensure
+     ActiveMerchant::Billing::Base.mode = original_base_mode
+  end
+
+  def test_setting_test_option_does_not_send_a_test_request
+    gateway = AuthorizeNetGateway.new(fixtures(:authorize_net).merge({:test => true}))
+    response = gateway.purchase(@amount, @credit_card, @options)
+
+    assert_success response
+    assert_equal('0', response.params['test_request'])
+  end
+
+  # you can send test requests to live servers
+  def test_setting_test_request_does_not_set_test_mode
+    original_base_mode = ActiveMerchant::Billing::Base.mode
+    ActiveMerchant::Billing::Base.mode = :production
+
+    gateway = AuthorizeNetGateway.new(fixtures(:authorize_net).merge({:test_request => true}))
+
+    gateway.expects(:ssl_post).with(gateway.live_url, any_parameters, any_parameters)
+    gateway.purchase(@amount, @credit_card, @options)
```

Comment:
Ah, ok - this test name looks a little unclear. Would it be better as `setting_test_request_goes_to_live_url` ?